Add classname attr to the JUnit test case and add a flag to allow skipping test junit report

chizhg opened the pull request:
1. Add `classname` attribute to the JUnit test case to show on Spyglass, this will fix https://github.com/kubernetes-sigs/kubetest2/issues/107
2. For some tester implementations, they will be generating their own Junit results, in which cases the extra `Test` JUnit result will be redundant. So add an extra flag `--skip-test-junit-report` to allow skipping reporting the JUnit test result. This will partially fix https://github.com/kubernetes-sigs/kubetest2/issues/58.

amwat reviewed (commented):
I wouldn't want to mess with the name too much as it invalidates a lot of testgrid history.

ref: https://github.com/kubernetes/test-infra/pull/21180#issuecomment-793063005

I think filtering out the kubetest2 specific entries can be done by whatever is post processing it by many other means? (sorting and relying on capitalized alphabetical order is not the most ideal)

chizhg commented:
> I wouldn't want to mess with the name too much as it invalidates a lot of testgrid history.
> 
> ref: [kubernetes/test-infra#21180 (comment)](https://github.com/kubernetes/test-infra/pull/21180#issuecomment-793063005)
> 
> I think filtering out the kubetest2 specific entries can be done by whatever is post processing it by many other means? (sorting and relying on capitalized alphabetical order is not the most ideal)

That makes total sense. I've reverted the change in my PR and updated the commit message accordingly.

spiffxp commented on pkg/app/cmd.go:
```diff
@@ -199,6 +200,7 @@ func (o *options) bindFlags(flags *pflag.FlagSet) {
 	flags.BoolVar(&o.up, "up", false, "provision the test cluster")
 	flags.BoolVar(&o.down, "down", false, "tear down the test cluster")
 	flags.StringVar(&o.test, "test", "", "test type to run, if unset no tests will run")
+	flags.BoolVar(&o.skipTestJUnitReport, "skip-test-junit-report", false, "skip reporting the test result as a JUnit test case")
```
Is "test" a necessary distinction here? 

Vs "skip-junit-report", "skip reporting kubetest2 steps to junit.xml", etc

amwat replied:
IIUC the intention is to only skip adding the junit entry for the `kubetest2.Test()` step. build, up, down will still show.
It's not too ideal but since we have explicit tester binaries I can see use cases where it should be treated specially.

chizhg replied:
Thanks @amwat . Yes, that is the intention for this change.

amwat commented on pkg/app/cmd.go:
```diff
@@ -199,6 +200,7 @@ func (o *options) bindFlags(flags *pflag.FlagSet) {
 	flags.BoolVar(&o.up, "up", false, "provision the test cluster")
 	flags.BoolVar(&o.down, "down", false, "tear down the test cluster")
 	flags.StringVar(&o.test, "test", "", "test type to run, if unset no tests will run")
+	flags.BoolVar(&o.skipTestJUnitReport, "skip-test-junit-report", false, "skip reporting the test result as a JUnit test case")
```
```suggestion
	flags.BoolVar(&o.skipTestJUnitReport, "skip-test-junit-report", false, "skip reporting the test step as a JUnit test case. used when solely relying on the tester binary to generate it's own junit.")
```

chizhg replied:
Done.

amwat reviewed (commented):
/approve

spiffxp reviewed (approved):
/lgtm
